SqlInstall.ps1

```diff
@@ -0,0 +1,433 @@
+            $sid = $ntprincipal.Translate([System.Security.Principal.SecurityIdentifier])
+            $sidstr = $sid.Value.ToString()
+        }
+        catch {
+            $sidstr = $null
+        }
+        Write-Message -Level Verbose -Message "Account: $($accountToAdd)" -ForegroundColor DarkCyan
+        if ( [string]::IsNullOrEmpty($sidstr) ) {
+            Write-Message -Level Verbose -Message "Account not found!" -ForegroundColor Red
+            #exit -1
+        }
+
+        Write-Message -Level Verbose -Message "Account SID: $($sidstr)" -ForegroundColor DarkCyan
+        $tmp = ""
+        $tmp = [System.IO.Path]::GetTempFileName()
+        Write-Message -Level Verbose -Message "Export current Local Security Policy" -ForegroundColor DarkCyan
+        secedit.exe /export /cfg "$($tmp)" 
+        $c = ""
+        $c = Get-Content -Path $tmp
+        $currentSetting = ""
+        foreach ($s in $c) {
+            if ( $s -like "SeManageVolumePrivilege*") {
+                $x = $s.split("=", [System.StringSplitOptions]::RemoveEmptyEntries)
+                $currentSetting = $x[1].Trim()
+            }
+        }
+
+
+        if ( $currentSetting -notlike "*$($sidstr)*" ) {
+            Write-Message -Level Verbose -Message "Modify Setting ""Perform Volume Maintenance Task""" -ForegroundColor DarkCyan
+       
+            if ( [string]::IsNullOrEmpty($currentSetting) ) {
+                $currentSetting = "*$($sidstr)"
+            }
+            else {
+                $currentSetting = "*$($sidstr),$($currentSetting)"
+            }
+       
+            Write-Message -Level Verbose -Message "$currentSetting"
+       
+            $outfile = @"
+[Unicode]
+Unicode=yes
+[Version]
+signature="`$CHICAGO`$"
+Revision=1
+[Privilege Rights]
+SeManageVolumePrivilege = $($currentSetting)
+"@
+       
+            $tmp2 = ""
+            $tmp2 = [System.IO.Path]::GetTempFileName()
+       
+       
+            Write-Message -Level Verbose -Message "Import new settings to Local Security Policy" -ForegroundColor DarkCyan
+            $outfile | Set-Content -Path $tmp2 -Encoding Unicode -Force
+            #notepad.exe $tmp2
+            Push-Location (Split-Path $tmp2)
+       
+            try {
+                secedit.exe /configure /db "secedit.sdb" /cfg "$($tmp2)" /areas USER_RIGHTS 
+                #Write-Message -Level Verbose -Message "secedit.exe /configure /db ""secedit.sdb"" /cfg ""$($tmp2)"" /areas USER_RIGHTS "
+            }
+            finally {  
+                Pop-Location
+            }
+        }
+        else {
+            Write-Message -Level Verbose -Message "NO ACTIONS REQUIRED! Account already in ""Perform Volume Maintenance Task""" -ForegroundColor DarkCyan
+        }
+        Write-Message -Level Verbose -Message "Done." -ForegroundColor DarkCyan 
+    }
+
+    
+
+    #Now configure the right amount of TempDB files.
+
+    $val = 1
+
+    WHILE ($val -ne $NumberOfCores) {
```

Comment:
lower case too, just for PowerShell, the SQL is good

Reply:
I hope I've got all of them...